gallery/src/components/NxTextLink/NxTextLinkPage.tsx

```diff
@@ -26,80 +26,125 @@ export default function NxTabsPage() {
+                  the link text will be appended with an "external link" icon. This prop also affects the defaults of
+                  the <NxCode>noReferrer</NxCode> and <NxCode>newTab</NxCode> props.
+                </NxTable.Cell>
+              </NxTable.Row>
+              <NxTable.Row>
+                <NxTable.Cell>noReferrer</NxTable.Cell>
+                <NxTable.Cell>boolean</NxTable.Cell>
+                <NxTable.Cell>No</NxTable.Cell>
+                <NxTable.Cell>false for internal links, true for external links</NxTable.Cell>
+                <NxTable.Cell>
+                  When set to true, the <NxCode>noreferrer</NxCode> rel is added to the link which
+                  prevents certain properties from being passed through to the link target that can allow the target
+                  to discern what site was linked from, specifically, the <NxCode>Referer</NxCode> HTTP
+                  header and the <NxCode>window.opener</NxCode> JavaScript property.
+                </NxTable.Cell>
+              </NxTable.Row>
+              <NxTable.Row>
+                <NxTable.Cell>newTab</NxTable.Cell>
+                <NxTable.Cell>boolean</NxTable.Cell>
+                <NxTable.Cell>No</NxTable.Cell>
+                <NxTable.Cell>false for internal links, true for external links</NxTable.Cell>
+                <NxTable.Cell>
+                  Whether or not this link should open in a new tab/window. Note that this is accomplished via the
+                  link's <NxCode>target</NxCode> attribute, and any explicitly
+                  set <NxCode>target</NxCode> will override this prop.
+                </NxTable.Cell>
+              </NxTable.Row>
+              <NxTable.Row>
+                <NxTable.Cell>truncate</NxTable.Cell>
+                <NxTable.Cell>boolean</NxTable.Cell>
+                <NxTable.Cell>No</NxTable.Cell>
+                <NxTable.Cell>false</NxTable.Cell>
+                <NxTable.Cell>
+                  When set, the link becomes a block element that constrains its text content to one line, and truncates
+                  the text with an ellipsis when necessary. If used in conjunction with <NxCode>external</NxCode>, the
+                  external link icon will always be visible, with the text ellipsis truncation occurring before the
+                  icon.
+                </NxTable.Cell>
+              </NxTable.Row>
+              <NxTable.Row>
+                <NxTable.Cell>disabled</NxTable.Cell>
+                <NxTable.Cell>boolean</NxTable.Cell>
+                <NxTable.Cell>No</NxTable.Cell>
+                <NxTable.Cell>false</NxTable.Cell>
+                <NxTable.Cell>
+                  When set, the text link is rendered as disabled and has <NxCode>aria-disabled="true"</NxCode>.
+                  <NxWarningAlert>
+                    Note: For accessibility purpose, the <NxCode>href</NxCode> attribute should be removed
```

Comment:
We shouldn't force the caller to manage this. Within `NxTextLink` we should determine, based on the `disabled` prop, whether to pass on the `href` and whether to set the `role`

Reply:
Updated here 9cf857091b7ef596d7235e54df4f271840f9a097.

In the Internal/ external link example code, I keep the 'href' attribute on the disabled text link just for demo purpose that it will be removed when the component is rendered. But let me know if you think it should be removed, thanks.